### PR TITLE
fix: do not throw while evaluating flag, return proper error

### PR DESCRIPTION
**Pull request by vahidlazio**

### CODE REVIEW

File: Confidence/src/main/java/com/spotify/confidence/ConfidenceFlagEvaluation.kt [hunk 1]
```diff
@@ -19,9 +16,10 @@ fun <T> FlagResolution?.getEvaluation(
     }
     requireNotNull(this)
     val resolvedFlag = this.flags.firstOrNull { it.flag == parsedKey.flagName }
-        ?: throw FlagNotFoundError(
-            "Could not find flag named: ${parsedKey.flagName}",
-            parsedKey.flagName
+        ?: return Evaluation(
```
Comment (nicklasl) [inline Confidence/src/main/java/com/spotify/confidence/ConfidenceFlagEvaluation.kt:19]: Do we need to check for this and throw in the provider instead now?

Comment (vahidlazio) [reply Confidence/src/main/java/com/spotify/confidence/ConfidenceFlagEvaluation.kt:19]: probably, but then I was thinking what's the point of throwing of OF case? the sdk will just handle it and send default anyways. right?

Comment (nicklasl) [reply Confidence/src/main/java/com/spotify/confidence/ConfidenceFlagEvaluation.kt:19]: I think the SDK expects the provider to throw for this case.

Comment (nicklasl) [reply Confidence/src/main/java/com/spotify/confidence/ConfidenceFlagEvaluation.kt:19]: Never mind. The SDK can support that the provider throws but it does not require it.

**Review by nicklasl — APPROVED** [review]

**Pull request merged**